scripts/create-package.js

```diff
@@ -125,35 +174,45 @@ function writeFileCallback(error, writeFileError, callback) {
 
 function updateRush() {
   // don't add the same package to rush.json twice
-  if (rushContainsPackage()) {
-    console.log(rushPackagePresent);
+  if (rushJson.projects.some(project => project.packageName === `@uifabric/${newPackageName}`)) {
+    console.error(rushPackagePresent);
     postRushUpdate();
     return;
   }
 
-  rushJson.projects.push({
-    packageName: `@uifabric/${newPackageName}`,
-    projectFolder: `packages/${newPackageName}`,
-    // versionPolicyName: 'lockedMajor',
-    shouldPublish: false
-  });
-
-  fs.writeFile(rushPath, JSON.stringify(rushJson, null, 2), error => {
-    if (!handleError(error, errorUnableToUpdateRush)) {
-      return;
-    }
-
+  // Hack to preserve the comments in rush.json: look for the sequence of closing braces/brackets
```

Comment:
Not a huge fan of this...

Comments are nice, but json is not parsable with them and it's entirely unpredictable to do this. I have written scripts which use rush.json to operate on packages in the repo, and I would be pretty confused why it works for some json files and not others.

I'd prefer to just remove the comments or move them to a readme or anything... similar reasoning to why package.jsons don't have comments next to their dependencies.

Reply:
Latest version removes this hack and uses `jju` for parsing and updates instead. (I tried `jsonc-parser` first, since it's made by Microsoft and used by VS Code...but despite more or less *claiming* to be able to update arrays, at runtime it gives an error that array updates aren't supported yet.)